.github/workflows/rust.yml

```diff
@@ -36,8 +41,8 @@ jobs:
   rs-benchmarks:
     name: Rust Criterion benchmarks
 
-    # Only save the rs-benchmarks job when a commit gets pushed to main
-    if: ${{ github.event_name == 'push' }}
+    # Only run the rs-benchmarks job when a commit gets pushed to main or we manually specify it
+    if: ${{ github.ref == 'refs/heads/main' || github.event.inputs.force_run_benchmarks }
```

Comment:
Using ref == 'refs/heads/main' is a bit more inline with the actual spirit than any push.